engine/spec.go

```diff
@@ -34,8 +39,9 @@ type (
 		Privileged   bool              `json:"privileged,omitempty"`
 		Resources    Resources         `json:"resources,omitempty"`
 		Pull         PullPolicy        `json:"pull,omitempty"`
-		RunPolicy    RunPolicy         `json:"run_policy,omitempty"`
+		RunPolicy    runtime.RunPolicy `json:"run_policy,omitempty"`
 		Secrets      []*SecretVar      `json:"secrets,omitempty"`
+		SpecSecrets  []*Secret         `json:"spec_secrets,omitempty"`
```

Comment:
This is a new field where we place the secret values themselves that are exposed through the step. This is to implement the `runtime.Secret` interface so that the common `Execer` can make the right log filter.